.github/workflows/build-and-test.yml

```diff
@@ -68,9 +68,32 @@ jobs:
       - name: gofmt
         run: |
           make install-tools
-          gofmt -s -w .
+          make fmt
           if ! git diff --exit-code; then
-            echo "One or more Go files are not formatted correctly. Run 'gofmt -s -w .' and push the changes."
+            echo "One or more Go files are not formatted correctly. Run 'make fmt' and push the changes."
+            exit 1
+          fi
+
+  generate-metrics:
+    name: generate-metrics
+    # Use 20.04.5 until https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/16450 is resolved
+    runs-on: ubuntu-20.04
+    needs: [setup-environment]
+    steps:
+      - name: Check out the codebase.
+        uses: actions/checkout@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: ${{ env.GO_VERSION }}
+          cache-dependency-path: '**/go.sum'
+
+      - name: generate-metrics
+        run: |
+          make generate-metrics
+          if ! git diff --exit-code; then
```

Comment:
fieldalignment updated the databricks receiver MetricsBuilder and this didn't fail.

Reply:
It originally did (see https://github.com/signalfx/splunk-otel-collector/actions/runs/4767265273/jobs/8475311980) before adding `make fmt` to the `generate-metrics` target.

Reply:
So make fmt would then cause a git diff though right? If we aren't committing the fieldalignment changes I think I missing something.

Reply:
I think that `internal/receiver/databricksreceiver/internal/metadata/generated_metrics.go` currently in main is correctly formatted.  But if we run `make generate-metrics` without `make fmt`, it is generated unformatted and causes a diff.

Reply:
sorry, I'm of course assuming our source wasn't fmt'ed.